From bd9b8a1f188381d3f05d3a85ea3650e70c36269f Mon Sep 17 00:00:00 2001
From: Tim Yao <tim.yao@amlogic.com>
Date: Fri, 20 Mar 2015 23:15:01 -0700
Subject: [PATCH 5500/5965] PD#104392 h265: I only play rate control.

Use fixed frame rate to make I only mode played more smoothly.

Change-Id: Ie750fabfa89caaf0a566c5a39d5670143926ef6a
---
 drivers/amlogic/amports/vh265.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vh265.c b/drivers/amlogic/amports/vh265.c
index 0c933c3c3a76..0184d4b6250b 100755
--- a/drivers/amlogic/amports/vh265.c
+++ b/drivers/amlogic/amports/vh265.c
@@ -3424,7 +3424,13 @@ static int prepare_display_buf(hevc_stru_t* hevc, PIC_t* pic)
             hevc->last_lookup_pts = vf->pts;
         }
 
-        if ((hevc->pts_mode == PTS_NONE_REF_USE_DURATION) && (slice_type != 2)) {
+        if (((hevc->pts_mode == PTS_NONE_REF_USE_DURATION) && (slice_type != 2)) ||
+            (i_only_flag)) {
+            if (frame_dur == 0) {
+                // add a default duration for 1/30 second if there is no valid frame duration available
+                frame_dur = 96000/30;
+            }
+
             vf->pts = hevc->last_pts + DUR2PTS(frame_dur);
         }
         hevc->last_pts = vf->pts;
-- 
2.19.0

